stan/math/prim/prob/lognormal_lpdf.hpp

```diff
@@ -85,7 +87,7 @@ return_type_t<T_y, T_loc, T_scale> lognormal_lpdf(const T_y& y, const T_loc& mu,
       stan::math::size(y));
   if (!is_constant_all<T_y>::value) {
     for (size_t n = 0; n < stan::math::size(y); n++) {
-      inv_y[n] = 1 / value_of(y_vec[n]);
+      inv_y[n] = 1 / value_of_rec(y_vec[n]);
```

Comment:
isn't it better to use "1.0 / ..." ? This way the expression is forced to be double.

Can you remind me of the difference of value_of and value_of_rec ?

Reply:
```value_of_rec``` will convert the int to a double, whereas ```value_of```` will just return the int. Not too sure about other differences beyond that, to be honest

Reply:
I'm not particularly married to either approach, so happy to change either way

Reply:
What about using `inv` (as you included the header anyway)?

Reply:
Also `value_of_rec` will not do what you expect for "nested" types like `fvar<var<double>`, as it would return `double`, while `value_of` would return `var<double>`. I haven't checked, but that may be at the bottom of the failure in the distribution tests.

Reply:
Ah good call, I'll make that change